components/lib/crash/src/main/java/mozilla/components/lib/crash/service/MozillaSocorroService.kt

```diff
@@ -20,54 +37,197 @@ typealias GeckoCrashReporter = CrashReporter
  *                [File a bug](https://bugzilla.mozilla.org/enter_bug.cgi?product=Socorro) if you would like to get your
  *                app added to the whitelist.
  */
+@Suppress("LargeClass")
 class MozillaSocorroService(
     private val applicationContext: Context,
     private val appName: String
 ) : CrashReporterService {
-    override fun report(crash: Crash.UncaughtExceptionCrash) {
-        // Not implemented currently.
+    private val logger = Logger("mozac/MozillaSocorroCrashHelperService")
+    private val startTime = System.currentTimeMillis()
+    private val defaultServerUrl = "https://crash-reports.mozilla.com/submit?" +
+            "id=${BuildConfig.MOZ_APP_ID}&version=${BuildConfig.MOZILLA_VERSION}&${BuildConfig.MOZ_APP_BUILDID}"
+    val partSet = mutableSetOf<String>()
 
-        // In theory we could upload uncaught exception crashes to Socorro too. But Socorro is not the best tool for
-        // them and it is not used by the app teams.
+    override fun report(crash: Crash.UncaughtExceptionCrash) {
+        sendReport(crash.throwable, null, null)
     }
 
     override fun report(crash: Crash.NativeCodeCrash) {
-        // GeckoView comes with a crash reporter class that we can use here. For now we are assuming that we only want
-        // to upload native code crashes to Socorro if GeckoView is used. If this assumption is going to be wrong in
-        // the future then we can start inlining the functionality here.
-        sendViaGeckoViewCrashReporter(crash)
+        sendReport(null, crash.minidumpPath, crash.extrasPath)
+    }
+
+    @Suppress("ComplexMethod", "LongMethod")
```

Comment:
We may be able to split this method into sub methods that add headers, send the meta data, send the crash data..? :)

Reply:
Agreed.